### PR TITLE
feat: add git hooks to format files pre commit

### PR DESCRIPTION
**Pull request by majordwarf**
Added execution of `Prettier` on every commit reducing manual workload.

**How to test:**
- ~Setup git hooks by executing `./scripts/setup-git-hooks`~
- Hooks now setup automatically as yarn `postInstall` lifecycle
- Make some changes to a file
- Track and commit the file

### CODE REVIEW

**Review by ryuk156 — DISMISSED** [review]
@majordwarf it works fine you can merge this PR

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# Fetch added, copied, modified and renamed files
+FILES=$(git diff --cached --name-only --diff-filter=ACMR --ignore-unknown | sed 's| |\\ |g')
```
Comment (endurance21) [inline .githooks/pre-commit]: `--ignore-unknown` seems new , cant find a doc for the same :confused: 
is this intention to add ? 
@majordwarf

Comment (endurance21) [reply .githooks/pre-commit]: i just tested this one , not working properly.
removing `ignore-unknown` works fine :question:

**Comment by pollend** [comment]
why do you want to have git introduce automatic changes before you commit anything. this is confusing at worst since you don't expect it to happen or it could be problematic at worse and introduce a breaking change. you also want to avoid making the scope of a change too large then it also makes it hard to review the relevant change. I would rather then be warned through ESLint with jenkins or github actions. 

@Cervator would be an asset here.

**Review by endurance21 — APPROVED** [review]

**Pull request closed without merging**